app/models/card.rb

```diff
@@ -1,7 +1,29 @@
 class Card < ActiveRecord::Base
-  before_validation :normalize_fields, :set_default_review_date
+  before_validation :normalize_fields, :set_new_review_date
   validates :original_text, :translated_text, :review_date, presence: true
   validate :not_equal_fields
+  scope :for_review, -> { where("review_date < ?", Time.now).order("RANDOM()") }
+
+  def move_review_date
+    set_new_review_date
+    save
+    Rails.cache.write('number_of_cards_for_review', Card.for_review.count)
+  end
+
+  def check_translation?(translation)
+    if translated_text == clear(translation)
+      move_review_date
+      return true
+    else
+      return false
+    end
+  end
+
+  def self.cards_for_review_counter
+    Rails.cache.fetch('number_of_cards_for_review') do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.